### ISSUE TITLE
Add USAC EstimationMethod::AffinePartial2D

### ISSUE THREAD

**Issue by llschloesser** [opencv/opencv#19166]
##### System information (version)
- OpenCV => 4.5
- Operating System / Platform => Ubuntu 20.04
- Compiler => g++ 9.3

##### Detailed description
A recent [GSOC contribution](https://docs.opencv.org/master/d1/df1/md__build_master-contrib_docs-lin64_opencv_doc_tutorials_calib3d_usac.html) added USAC options to many existing geometry functions. However, USAC support was not added to `cv::estimateAffinePartial2D` or `cv::estimateAffine2D` (though an overload was created for the latter).

See: `enum EstimationMethod { Homography, Fundamental, Fundamental8, Essential, Affine, P3P, P6P};`

This is a feature request to add USAC support to the previously existing functions listed above, as well as a new overload for partial affine 2D or similarity transforms.

##### Steps to reproduce
N/A (or simply try to pass `cv::MAGSAC` to `cv::estimateAffinePartial2D`)

##### Issue submission checklist
 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by vestri** [comment]
Hi,
it would be nice if you can expose usac so that we can use it for solving other estimation problems (see #17316).
thanks

**Comment by asmorkalov** [comment]
@ivashmak could you take a look?

**Comment by ivashmak** [comment]
Yes, USAC does not have AffinePartial2D (4 DoF) but has Affine2D (6 DoF). I think it should not take too long to implement.

**Comment by ArcticLampyrid** [comment]
This is still missing. Has there been any progress?
